pkg/console/controllers/oauthclients/oauthclients.go

```diff
@@ -158,9 +163,32 @@ func (c *oauthClientsController) sync(ctx context.Context, controllerContext fac
 			return c.statusHandler.FlushAndReturn(oauthErr)
 		}
 		return nil
-	case "OIDC":
+
+	case configv1.AuthenticationTypeOIDC, configv1.AuthenticationTypeNone:
 		c.oauthClientsInformerSwitch.Stop()
+
+		oidcConfig := utilsub.GetOIDCClientConfig(authnConfig)
+		if oidcConfig == nil {
+			klog.Error("oidc client configuration not found")
+			return nil
+		}
+
+		clientSecret, err := c.configNSSecretsLister.Secrets(api.OpenShiftConfigNamespace).Get(oidcConfig.ClientSecret.Name)
+		if apierrors.IsNotFound(err) {
+			klog.Error("oauth client config secret not found")
```

Comment:
if the config exists, points to a secret but the secret does not exist - that would be a good place to go Degraded=true